Closes #29 [devdocs] [references] Hexdumps in references

stdevEdu opened the pull request.

stdevEdu committed:
[devdocs] [references] refs #30 Fixed file structure
Merged master changes

stdevEdu committed:
[devdocs] [references] refs #29 #31 Messages hexdumps in P2P reference

olemis commented on content/devdocs/references/p2p_networking.md:
````diff
@@ -71,7 +71,12 @@ The following example is an annotated hex dump of a mainnet message
-TODO : GETP message hex dump
+GetPeersMessage:
+```
+04 00 00 00 00 00 00 00 ........................... Full message
````
No need to add full message . It's too much , especially if messages are long. Annotated message would be enough. Please remove

olemis commented on content/devdocs/references/p2p_networking.md:
````diff
@@ -136,7 +197,15 @@ The following annotated hexdump shows a `GETB` message.  (The
-TODO : GETB message hex dump
+GetBlocksMessage:
+```
+14 00 00 00 00 00 00 00 d2 04 00 00 00 00 00 00 
````
No need to add full message . It's too much , especially if messages are long. Annotated message would be enough. Please remove

olemis commented on content/devdocs/references/p2p_networking.md:
````diff
@@ -136,7 +197,15 @@ The following annotated hexdump shows a `GETB` message.  (The
+GetBlocksMessage:
+```
+14 00 00 00 00 00 00 00 d2 04 00 00 00 00 00 00 
+2e 16 00 00 00 00 00 00 ........................... Full message
````
No need to add full message . It's too much , especially if messages are long. Annotated message would be enough. Please remove

olemis commented on content/devdocs/references/p2p_networking.md:
````diff
@@ -108,7 +113,63 @@ The following annotated hexdump shows a `GIVB` message.  (The
-TODO : GIVB message hex dump
+GiveBlocksMessage:
+```
+8a 01 00 00 00 00 00 00 02 00 00 00 00 00 00 00 
````
No need to add full message . It's too much , especially if messages are long. Annotated message would be enough. Please remove

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -191,7 +260,29 @@ The following annotated hexdump shows an `ANNB` message with two
+------------------------------------------------------------------------
+0x0000 0c 00 00 00 00 00 00 00 ........................... Prefix
+0x0008 40 e2 01 00 00 00 00 00 ........................... MaxBkSeq
+GetTxns:
```
Oops ! Remove this message

olemis commented on content/devdocs/references/p2p_networking.md:
````diff
@@ -420,7 +626,16 @@ message header has been omitted and the actual IP addresses have been
-TODO : INTR message hex dump
+IntroductionMessage:
+```
+0e 00 00 00 00 00 00 00 d2 04 00 00 d2 1e 05 00
````
No need to add full message . It's too much , especially if messages are long. Annotated message would be enough. Please remove

olemis commented on content/devdocs/references/p2p_networking.md:
````diff
@@ -315,7 +511,17 @@ message header has been omitted and the actual IP address has been
+GivePeersMessage:
+```
+1a 00 00 00 00 00 00 00 03 00 00 00 5d 87 b2 76 
+70 17 9c 21 58 2f 70 17 94 67 29 79 70 17 ......... Full message
````
No need to add full message . It's too much , especially if messages are long. Annotated message would be enough. Please remove

olemis commented on includes/references.md:
```diff
@@ -30,47 +29,41 @@ http://opensource.org/licenses/MIT.
 [skycoin-docs mailing list]: TODO: EMPTY
-[skycoin.go at master]: TODO: EMPTY
-[skycoin.go at testnet]: TODO: EMPTY
-[standard block relay]: /en/developer-guide#term-standard-block-relay "The regular block relay method: announcing a block with an ANNB message and waiting for a response"
```
keep this reference please

olemis commented on includes/references.md:
```diff
@@ -30,47 +29,41 @@ http://opensource.org/licenses/MIT.
 [network]: /en/developer-guide#term-network "The Skycoin gnet P2P network which broadcasts transactions and blocks"
 [not a specification]: /en/developer-reference#not-a-specification
-[not aspecification]: TODO: EMPTY
 [proof of burn]: TODO: EMPTY
```
Document this TODO

olemis commented on includes/references.md:
```diff
@@ -30,47 +29,41 @@ http://opensource.org/licenses/MIT.
-[introduction handshake]: TODO: EMPTY
+[introduction handshake]:  https://en.wikipedia.org/wiki/Handshaking "A handshake is an automated process of negotiation between two communicating participants that establishes the protocols of a communications channel, at the start of the communication, before full communication begins. It follows the physical establishment of the channel and precedes normal information transfer."
 [maximum message length]: TODO: EMPTY
 [miner]: TODO: EMPTY
```
Missing

olemis commented on includes/references.md:
```diff
@@ -30,47 +29,41 @@ http://opensource.org/licenses/MIT.
-[intr messages]: TODO: EMPTY
-[introduction handshake]: TODO: EMPTY
+[introduction handshake]:  https://en.wikipedia.org/wiki/Handshaking "A handshake is an automated process of negotiation between two communicating participants that establishes the protocols of a communications channel, at the start of the communication, before full communication begins. It follows the physical establishment of the channel and precedes normal information transfer."
 [maximum message length]: TODO: EMPTY
```
Missing

olemis commented on includes/references.md:
```diff
@@ -5,7 +5,6 @@ http://opensource.org/licenses/MIT.
 {% endcomment %}
 
 
-[annb message]: TODO: EMPTY
```
Bring this back

olemis replied:
This change is still needed . Please check all other changes have been implemented before merging this PR

olemis commented on includes/references.md:
```diff
@@ -5,7 +5,6 @@ http://opensource.org/licenses/MIT.
 
 
-[annb message]: TODO: EMPTY
 [annt message]: TODO: EMPTY
```
Missing

olemis commented on includes/references.md:
```diff
@@ -30,47 +29,41 @@ http://opensource.org/licenses/MIT.
 [high-speed block relay network]: https://www.mail-archive.com/bitcoin-development@lists.sourceforge.net/msg03189.html
 [intr message]: TODO: EMPTY
-[intr messages]: TODO: EMPTY
-[introduction handshake]: TODO: EMPTY
```
Bring this back

olemis commented on includes/references.md:
```diff
@@ -30,47 +29,41 @@ http://opensource.org/licenses/MIT.
 [bitcoin verack message]: https://bitcoin.org/en/developer-reference#verack "A P2P network message sent in reply to a version message to confirm a connection has been established"
 [bitcoin version message]: https://bitcoin.org/en/developer-reference#version "A P2P network message sent at the begining of a connection to allow protocol version negotiation"
-[block height]: TODO: EMPTY
+[block height]: /docs/skycoin/_data/glossary/en/block-height.yaml
```
nope ... please follow instructions in README.md file to run the Hugo web site and insert the right URL in here

olemis commented on content/devdocs/references/p2p_networking.md:
````diff
@@ -191,7 +260,29 @@ The following annotated hexdump shows an `ANNB` message with two
-TODO: ANNB message hex dump
+AnnounceBlocksMessage:
+```
+0c 00 00 00 00 00 00 00 40 e2 01 00 00 00 00 00 
````
No need to add full message . It's too much , especially if messages are long. Annotated message would be enough. Please remove

stdevEdu committed:
[devdocs] [references] refs #29 #31 Fixed GETT hex misplacing error

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -191,7 +260,29 @@ The following annotated hexdump shows an `ANNB` message with two
+0c 00 00 00 00 00 00 00 40 e2 01 00 00 00 00 00 
+................................................... Full message
+------------------------------------------------------------------------
+0x0000 0c 00 00 00 00 00 00 00 ........................... Prefix
```
This is not quite like it should be . First four bytes are the total size of the buffer that follows . The other four bytes are the message `Prefix` ID (e.g.one of `INTR`, `GETB`, ...) . Please fix that

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -218,7 +309,23 @@ The following annotated hexdump shows an `ANNT` message with two
+c2 23 c0 57 d2 bf 80 d3 f2 a3 29 82 c3 2c 3c 0d 
+b8 e2 67 48 20 da 50 64 78 3f b0 97 ............... Full message
+------------------------------------------------------------------------
+0x0000 48 00 00 00 00 00 00 00 ........................... Prefix
```
This is not quite like it should be . First four bytes are the total size of the buffer that follows . The other four bytes are the message `Prefix` ID (e.g.one of `INTR`, `GETB`, ...) . Please fix that

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -250,7 +357,96 @@ The following annotated hexdump shows a `GIVT` message.  (The
+53 7e 75 bb 72 07 22 00 00 00 00 00 00 00 38 00 
+00 00 00 00 00 00 ................................. Full message
+------------------------------------------------------------------------
+0x0000 82 02 00 00 00 00 00 00 ........................... Prefix
```
This is not quite like it should be . First four bytes are the total size of the buffer that follows . The other four bytes are the message `Prefix` ID (e.g.one of `INTR`, `GETB`, ...) . Please fix that

olemis commented on content/devdocs/references/p2p_networking.md:
````diff
@@ -315,7 +511,17 @@ message header has been omitted and the actual IP address has been
-GIVP message hex dump
+GivePeersMessage:
+```
+1a 00 00 00 00 00 00 00 03 00 00 00 5d 87 b2 76 
````
No need to add full message . It's too much , especially if messages are long. Annotated message would be enough. Please remove

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -315,7 +511,17 @@ message header has been omitted and the actual IP address has been
+1a 00 00 00 00 00 00 00 03 00 00 00 5d 87 b2 76 
+70 17 9c 21 58 2f 70 17 94 67 29 79 70 17 ......... Full message
+------------------------------------------------------------------------
+0x0000 1a 00 00 00 00 00 00 00 ........................... Prefix
```
This is not quite like it should be . First four bytes are the total size of the buffer that follows . The other four bytes are the message `Prefix` ID (e.g.one of `INTR`, `GETB`, ...) . Please fix that

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -420,7 +626,16 @@ message header has been omitted and the actual IP addresses have been
+0e 00 00 00 00 00 00 00 d2 04 00 00 d2 1e 05 00
+00 00 ............................................. Full message
+------------------------------------------------------------------------
+0x0000 0e 00 00 00 00 00 00 00 ........................... Prefix
```
This is not quite like it should be . First four bytes are the total size of the buffer that follows . The other four bytes are the message `Prefix` ID (e.g.one of `INTR`, `GETB`, ...) . Please fix that

olemis reviewed (changes requested):
In all cases for message dumps , please do the following:

- Put a pipe char with spaces (i.e. `   |   `) to separate offset out of hex dump
- Add a final offset at the end of the hex dump to easily see message size

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -108,7 +113,63 @@ The following annotated hexdump shows a `GIVB` message.  (The
+00 00 00 00 00 00 00 00 00 00 00 00 00 00 ......... Full message
+------------------------------------------------------------------------
+0x0000 8a 01 00 00 00 00 00 00 ........................... Prefix
+0x0008 02 00 00 00 ....................................... Blocks header
```
This here is the number of items of the subsequent blocks array , not a header

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -165,7 +234,23 @@ The following annotated hexdump shows a `GETT` message.  (The
+b8 e2 67 48 20 da 50 64 78 3f b0 97 ............... Full message
+------------------------------------------------------------------------
+0x0000 48 00 00 00 00 00 00 00 ........................... Prefix
+0x0008 02 00 00 00 ....................................... Txns header
```
This here is the number of items of the subsequent array , not a header

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -218,7 +310,23 @@ The following annotated hexdump shows an `ANNT` message with two
+b8 e2 67 48 20 da 50 64 78 3f b0 97 ............... Full message
+------------------------------------------------------------------------
+0x0000 48 00 00 00 00 00 00 00 ........................... Prefix
+0x0008 02 00 00 00 ....................................... Txns header
```
This here is the number of items of the subsequent array , not a header

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -250,7 +358,96 @@ The following annotated hexdump shows a `GIVT` message.  (The
+00 00 00 00 00 00 ................................. Full message
+------------------------------------------------------------------------
+0x0000 82 02 00 00 00 00 00 00 ........................... Prefix
+0x0008 02 00 00 00 ....................................... Txns header
```
This here is the number of items of the subsequent array , not a header

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -315,7 +512,17 @@ message header has been omitted and the actual IP address has been
+70 17 9c 21 58 2f 70 17 94 67 29 79 70 17 ......... Full message
+------------------------------------------------------------------------
+0x0000 1a 00 00 00 00 00 00 00 ........................... Prefix
+0x0008 03 00 00 00 ....................................... Peers header
```
This here is the number of items of the subsequent array , not a header

olemis reviewed (changes requested):
Please document array count accurately

stdevEdu committed:
[devdocs] [references] Fixed prefix error
Deleted full unindexed hexdumps

stdevEdu committed:
[devdocs] [references] refs #29 #31 Added final offset
Added pipe between offset and dump

olemis commented on content/devdocs/references/p2p_networking.md:
```diff
@@ -108,7 +113,39 @@ The following annotated hexdump shows a `GIVB` message.  (The
+0x0004 | 47 49 56 42 ....................................... Prefix
+0x0008 | 02 00 00 00 ....................................... Blocks header
+0x000c | 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
+       00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
```
@stdevEdu please add `<OFFSET> | ` prefix in all lines for readability so that hex dump columns be aligned . Fix that in your script

stdevEdu replied:
Done

olemis commented on content/devdocs/references/p2p_networking.md:
````diff
@@ -108,7 +113,39 @@ The following annotated hexdump shows a `GIVB` message.  (The
+```
+0x0000 | 8a 01 00 00 ....................................... Length
+0x0004 | 47 49 56 42 ....................................... Prefix
+0x0008 | 02 00 00 00 ....................................... Blocks header
````
Blocks header => Blocks length

stdevEdu replied:
Done

olemis reviewed (changes requested):
Almost there , but not quite yet

stdevEdu committed:
[devdocs] [references] refs #29 #31 Added offset to every line
Restored [annb message] line @ include/references.md
Changed "header" by "length"
